.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022, Kry10 Limited
+# SPDX-License-Identifier: BSD-2-Clause
+
+name: Build
+
+on:
+  pull_request:
+    # Currently aspirational
+    branches:
+      - master
+  push:
+    branches:
+      - ci-riscv64
+  schedule:
+    - cron: '0 17 * * 6'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout graph-refine
+        if: github.event_name != 'schedule'
+        uses: actions/checkout@v3
+      - name: Checkout graph-refine (schedule)
+        if: github.event_name == 'schedule'
+        uses: actions/checkout@v3
+        with:
+          ref: ci-riscv64
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: seL4
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Git
+        env:
+          CI_SSH: ${{ secrets.CI_SSH }}
+        shell: bash
+        run: |
+          # Configure SSH access and Git identity
+          eval $(ssh-agent)
+          ssh-add -q - <<< "${CI_SSH}"
+          echo "SSH_AUTH_SOCK=${SSH_AUTH_SOCK}" >> "${GITHUB_ENV}"
+          git config --global user.name "seL4 CI"
+          git config --global user.email "ci@sel4.systems"
+      - name: Install Nix
+        uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - name: Install Cachix
+        uses: cachix/cachix-action@v10
+        with:
+          name: sel4-bv
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      - name: Build graph-refine
+        run: nix-build -A graph-refine-image -o graph-refine-image nix/graph-refine.nix
```

Comment:
Can't say I'm a fan of introducing yet another build and configuration system. Happy to use this for now in the interest of getting things up and running, which is more important, but we should discuss pros and cons and overall policy on this when things are in a working state.